src/SSHDebugPS/ConnectionManager.cs

```diff
@@ -76,25 +89,31 @@ public static Connection GetSSHConnection(string name)
             if (connectionInfo == null)
             {
                 IVsConnectionManager connectionManager = (IVsConnectionManager)ServiceProvider.GlobalProvider.GetService(typeof(IVsConnectionManager));
-
-                string userName;
-                string hostName;
-
-                int atSignIndex = name.IndexOf('@');
-                if (atSignIndex > 0)
+                IConnectionManagerResult result;
+                if (string.IsNullOrWhiteSpace(name))
                 {
-                    userName = name.Substring(0, atSignIndex);
-                    hostName = name.Substring(atSignIndex + 1);
+                    result = connectionManager.ShowDialog();
                 }
                 else
                 {
-                    userName = string.Format(CultureInfo.CurrentCulture, StringResources.UserName_PlaceHolder);
-                    hostName = name;
-                }
+                    string userName;
+                    string hostName;
 
-                PasswordConnectionInfo newConnectionInfo = new PasswordConnectionInfo(hostName, userName, new System.Security.SecureString());
+                    int atSignIndex = name.IndexOf('@');
+                    if (atSignIndex > 0)
+                    {
+                        userName = name.Substring(0, atSignIndex);
+                        hostName = name.Substring(atSignIndex + 1);
```

Comment:
Is it possible for the user to supply ```user@``` and hostname would Index out of bounds?

Reply:
Added a check and will default to string.empty for the hostname in that case as that is what the user wanted.

Reply:
I think that is just invalid input.

---
In reply to: [287481980](https://github.com/microsoft/MIEngine/pull/868#discussion_r287481980) [](ancestors = 287481980)